tests/integration/test_lineage.py

```diff
@@ -103,48 +116,234 @@ def test_get_column_lineage(config, ctx):
+
+
+# test for cross-schema where the downstream is on reservations_schema and the upstreams are on hotel_schema
+@pytest.mark.db
+def test_cross_schema(testdata: Path, ctx):
+    scheme = "hana"  # make this into dictionary
+    username = "system"
+    password = "HXEHana1"
+    at = "localhost:39041"
+    db = "HXE"
+
+    # create dict for config
+    config_dict = {
+        "username": "system",
+        "password": "HXEHana1",
+        "host_port": "localhost:39041",
+        "database": "HXE",
+        "include_view_lineage": "true",
+        "include_column_lineage": "true",
+        "schema_pattern": AllowDenyPattern(allow=["RESERVATIONS*"]),
+    }
+
+    test_schema_file = testdata / "test_cross_schema.sql"
+    test_schema = test_schema_file.read_text()
+
+    uri = make_sqlalchemy_uri(scheme, username, password, at, db)
+    engine = create_engine(uri)
+    conn = engine.connect()
+
+    schema_results = conn.execute(
+        "SELECT * FROM SCHEMAS WHERE SCHEMA_NAME = 'RESERVATIONS_SCHEMA'"
+    )
+    print(schema_results)
+
+    if schema_results.returns_rows:
+        conn.execute("DROP SCHEMA RESERVATIONS_SCHEMA CASCADE")
+
+    conn.execute("CREATE SCHEMA RESERVATIONS_SCHEMA")
+    conn.execute(test_schema)
+
+    assert any(
+        "RESERVATIONS_SCHEMA" in item[0] for item in schema_results
+    ), f"{schema_results}"
+
+    # source class
+    source = HanaSource.create(config_dict, ctx)
+    source.get_db_connection()
+    inspector = inspect(conn)
+
+    view_definitions = source.get_column_lineage_view_definitions(inspector)
+
+    # sql = """SELECT
+    # H.NAME,
+    # R.TYPE,
+    # R.FREE,
+    # COUNT(R.FREE) AS FREE_RM_COUNT
+    # FROM
+    # HOTEL_SCHEMA.ROOM AS R
+    # LEFT JOIN
+    # HOTEL_SCHEMA.HOTEL AS H
+    # ON H.HNO = R.HNO
+    # GROUP BY
+    # H.NAME,
+    # R.TYPE,
+    # R.FREE;"""
+
+    for views in view_definitions:
+        # pass
+        assert views.name == "test_cross_schema"
+        assert views.schema == "reservations_schema"
+
+    lineages = list(source.get_column_view_lineage_elements(inspector))
+    assert len(lineages) == 1
+    cross_schema = lineages[0]
+    column_lineage = cross_schema[1]
+
+    # test for downstream cols
+    expected_downstreams = [
+        "name",
+        "type",
+        "free",
+        "free_rm_count",
+    ]
+    downstreams = [x[0].name for x in column_lineage]
+    assert (
+        downstreams == expected_downstreams
+    ), f"cross_schema_downstreams: {downstreams}"
+
+    # test for upstream cols
+    upstream_field_names = [
+        ["name"],  # name of hotel
+        ["type"],  # type of room
+        ["free"],  # status  of room
+        ["free"],  # status  of room
     ]
     # note that we sort the upstreams because the order is not guaranteed by sqlglot
-    upstreams = [sorted([source.name for source in x[1]]) for x in column_lineage]
+    upstreams = [sorted([source.name for source in x[1]])
+                 for x in column_lineage]
```

Comment:
```suggestion
    upstreams = [
        sorted([source.name for source in x[1]])
        for x in column_lineage
    ]
```

we generally prefer this style, adding a new line after the bracket